File: spring/src/main/java/com/projetos/model/Projeto.java
package com.projetos.model;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

@Data
@Entity
@Table(name = "projetos")
public class Projeto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String titulo;

    @Column(nullable = false)
    private String coordenador;

    @Column(nullable = false)
    private String curso;

    @Column(nullable = false)
    private Integer ano;

    @JdbcTypeCode(SqlTypes.VARCHAR)
    @Column(nullable = false)
    private String status = "ativo";
}

File: node/src/models/Projeto.js
const { DataTypes } = require('sequelize');
const sequelize = require('../database');

const Projeto = sequelize.define('Projeto', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
  },
  titulo: {
    type: DataTypes.STRING(255),
    allowNull: false,
  },
  coordenador: {
    type: DataTypes.STRING(255),
    allowNull: false,
  },
  curso: {
    type: DataTypes.STRING(255),
    allowNull: false,
  },
  ano: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  status: {
    type: DataTypes.ENUM('ativo', 'finalizado'),
    allowNull: false,
    defaultValue: 'ativo',
  },
}, {
  tableName: 'projetos',
  timestamps: false,
});

module.exports = Projeto;
